### DISCUSSION TITLE
Is there any way to get the app.stage from Graphic

### DISCUSSION THREAD

**Discussion by lykl** [pixijs/pixijs#9025] [Q&A]
Let's assume I can't get the `app` instance, how can I get the `app.stage`(I can get one `Graphic` instance)？
Is there an alternative to recursion?

**Comment by domis86** [comment]
You dont have access to `app` anywhere in your code? You cant pass it from somewhere else to where you need it?

**Reply by lykl** [reply]
Of course I can get `app`, but the place I need it is in a base class, it's difficult to get it

**Comment by SuperSodaSea** [comment]
```js
let stage = /* Your Graphics object */;
while (stage.parent)
    stage = stage.parent;
```

Works if your Graphics object has already been added to the stage.
